### PR TITLE
r/virtual_machine: Split disk Diff

### PR DESCRIPTION
**Pull request by bill-rich**
Split virtual disk diff function into validation and normalization. Some
normalization tasks should not be performed on new or unmatched virtual
disks, but validation should be performed on all.

### CODE REVIEW

**Comment by vancluever** [comment]
Hey @bill-rich, the problems that we talked about with the diff validation on new disks is happening because the entire diff validation is happening during the stage where we are comparing resources in state with the current state of configuration, remember. I noticed that `ValidateDiff` still has some values in it that rely on old state, and the operations are executed one after the other, so I'm not too sure if this will fix things.

Also, I was thinking that `NormalizeDiff` is probably not the best name for the second function, and possibly `ValidateDiff` is not necessarily either as while it is accurate it kind of assumes that we are only validating once when we are validating more times. Maybe some different naming is in order that reflects these realities?

We should also throw in some `PlanOnly` tests to make sure that validation is functional in the stages we are trying to target here (diffs with new disks being added, so either new resources completely or diffs where we are adding disks).

Thanks!

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 1]
```diff
@@ -379,6 +379,46 @@ func TestAccResourceVSphereVirtualMachine_highDiskUnitInsufficientBus(t *testing
...
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceVSphereVirtualMachineConfigBasic(),
+				Check:  testAccResourceVSphereVirtualMachineCheckExists(true),
+			},
+			{
+				Config:      testAccResourceVSphereVirtualMachineConfigMultiHighBusInsufficientBus(),
+				ExpectError: regexp.MustCompile("unit_number on disk \"disk1\" too high \\(15\\) - maximum value is 14 with 1 SCSI controller\\(s\\)"),
```
Comment (vancluever) [inline vsphere/resource_vsphere_virtual_machine_test.go]: Can this be in the first step or is there a reason why the VM needs to be created first?

Comment (bill-rich) [reply vsphere/resource_vsphere_virtual_machine_test.go]: That one is included to make sure the diff validation is working on updates and when there is already a disk present (disk0). This same case is tested at VM creation at: https://github.com/terraform-providers/terraform-provider-vsphere/blob/8027f5e805e92851893755bffa34be1efb9cb2c0/vsphere/resource_vsphere_virtual_machine_test.go#L362

The two tests added in this commit may be a bit overkill as there isn't a specific past bug that I'm trying to check for, but trying to cover other cases where the behavior is slightly different.

Comment (bill-rich) [reply vsphere/resource_vsphere_virtual_machine_test.go]: I dropped the two update tests, so its just the original `TestAccResourceVSphereVirtualMachine_highDiskUnitInsufficientBus` again.

**Pull request merged**